app/src/main/java/de/westnordost/streetcomplete/quests/air_pump/AddBicyclePump.kt

```diff
@@ -0,0 +1,45 @@
+package de.westnordost.streetcomplete.quests.air_pump
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment
+import de.westnordost.streetcomplete.util.ktx.toYesNo
+import de.westnordost.streetcomplete.osm.updateWithCheckDate
+
+class AddBicyclePump : OsmFilterQuestType<Boolean>() {
+
+    override val elementFilter = """
+        nodes, ways with
+        amenity = bicycle_repair_station or shop = bicycle
+        and (
+            !compressed_air and !service:bicycle:pump
+            or service:bicycle:pump older today -6 years
+        )
+        and access !~ private|no
+    """
+    override val changesetComment = "Add whether bicycle pump is available"
+    override val wikiLink = "Key:service:bicycle:pump"
+    override val icon = R.drawable.ic_quest_air_pump
+    override val questTypeAchievements = listOf(BICYCLIST)
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_air_pump_bicycle_title
+
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter("""
+            nodes, ways with
+            compressed_air
+            or service:bicycle:pump
+            or amenity ~ compressed_air|fuel|bicycle_repair_station
```

Comment:
same as for the other... though assuming that if not specified otherwise, a bicycle repair station has a working pump. (you may remove this if you think this cannot be assumed)
```suggestion
            nodes, ways with
            service:bicycle:pump = yes
            or amenity ~ bicycle_repair_station and service:bicycle:pump != no
```

Reply:
> though assuming that if not specified otherwise, a bicycle repair station has a working pump

this one was actually added after I proposed it: it was intended to help in cases of places with multiple repair stations by showing all of them for better orientation

Reply:
The highlighting must be consistent. Then I suggest to highlight exactly what the quest filter is asking for: any bicycle shops and any bicycle repair stations, regardless whether they have compressed air, a pump or not

Reply:
It makes sense to me to show it on _"any bicycle shops and any bicycle repair stations"_ as you say @westnordost, but I'd **_also_** highlight it on _nearby_ places which are **_explicitly_** marked as having compressed air or a bicycle pump, even if they are marked on some **_other_** nearby feature - otherwise the mapper could easily confuse one for another. Those happen relatively often: for example air pumps marked on [bicycle parkings](https://www.openstreetmap.org/node/8452855617), [vending machines](https://www.openstreetmap.org/node/1610293356), [bicycle rentals](https://www.openstreetmap.org/way/431729735), [picnic shelters](https://www.openstreetmap.org/node/8754095941), [cafes](https://www.openstreetmap.org/node/7200927563), [sport centers](https://www.openstreetmap.org/way/277433336) and [many more](https://overpass-turbo.eu/s/1hlj).

In that light, would something like 7ff7f4d2841e9969b18e89d0802713d18f281bf9 then be acceptable solution?

Reply:
ok